consensus/polybft/stake_manager.go

```diff
@@ -353,24 +333,29 @@ func newValidatorStakeMap(validatorSet AccountSet) validatorStakeMap {
 	return stakeMap
 }
 
-// setStake sets given amount of stake to a validator defined by address
-func (sc *validatorStakeMap) setStake(address types.Address, amount *big.Int) {
-	isActive := amount.Cmp(bigZero) > 0
-
+// addStake adds given amount to a validator defined by address
+func (sc *validatorStakeMap) addStake(address types.Address, amount *big.Int) {
 	if metadata, exists := (*sc)[address]; exists {
-		metadata.VotingPower = amount
-		metadata.IsActive = isActive
+		metadata.VotingPower.Add(metadata.VotingPower, amount)
+		metadata.IsActive = metadata.VotingPower.Cmp(bigZero) > 0
 	} else {
 		(*sc)[address] = &ValidatorMetadata{
 			VotingPower: new(big.Int).Set(amount),
 			Address:     address,
-			IsActive:    isActive,
+			IsActive:    amount.Cmp(bigZero) > 0,
 		}
 	}
 }
 
-// getActiveValidators returns all validators (*ValidatorMetadata) in sorted order
-func (sc validatorStakeMap) getActiveValidators(maxValidatorSetSize int) AccountSet {
+// removeStake removes given amount from validator defined by address
+func (sc *validatorStakeMap) removeStake(address types.Address, amount *big.Int) {
+	stakeData := (*sc)[address]
+	stakeData.VotingPower.Sub(stakeData.VotingPower, amount)
+	stakeData.IsActive = stakeData.VotingPower.Cmp(bigZero) > 0
+}
+
+// getSorted returns all validators (*ValidatorMetadata) in sorted order
```

Comment:
Maybe reword it like this, because it depends on `maxValidatorSetSize` how many are returned.
```suggestion
// getSorted returns validators (*ValidatorMetadata) in sorted order
```